src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -189,9 +189,14 @@ static bool HasOrderByHllType(List *sortClauseList, List *targetList);
  * Third, the function pulls up the collect operators in the tree. Fourth, the
  * function finds the extended operator node, and splits this node into master
  * and worker extended operator nodes.
+ *
+ * We also pass plannerRestrictionContext to the optimizer. The context is not
+ * used for any distributed optimization purposes. Instead, it is used to produce
+ * meaningful error messages for subquery pushdown planning.
```

Comment:
Besides `meaningful `, we actually use it to decide if we can push down the subquery. Maybe we can also add this to the comment.